pkg/gateway/api/api_grpc.go

```diff
@@ -349,6 +349,88 @@ func (s *grpcGatewayService) validateGetEvaluationsRequest(req *gwproto.GetEvalu
 	return nil
 }
 
+/*
+getPrerequisiteDownwards gets the features specified as prerequisite by the targetFeatures.
+*/
+func (s *grpcGatewayService) getPrerequisiteDownwards(
+	targetFeatures, allFeatures map[string]*featureproto.Feature,
+) (map[string]*featureproto.Feature, error) {
+	prerequisites := make(map[string]*featureproto.Feature, 0)
+	// depth first search
+	queue := make([]*featureproto.Feature, 0)
+	for _, f := range targetFeatures {
+		queue = append(queue, f)
+	}
+	for len(queue) > 0 {
+		f := queue[0]
+		for _, p := range f.Prerequisites {
+			preFeature, ok := allFeatures[p.FeatureId]
+			if !ok {
+				return nil, ErrFeatureNotFound
+			}
+			prerequisites[preFeature.Id] = preFeature
+			queue = append(queue, preFeature)
+		}
+		queue = queue[1:]
+	}
+	if len(prerequisites) == 0 {
+		return targetFeatures, nil
+	}
+	return s.mapMerge(targetFeatures, prerequisites), nil
+}
+
+/*
+getPrerequisiteUpwards recursively gets the features that have the specified targetFeatures as the prerequisite.
+*/
+func (s *grpcGatewayService) getPrerequisiteUpwards( // nolint:unused
+	targetFeatures, featuresHavePrerequisite map[string]*featureproto.Feature,
+) (map[string]*featureproto.Feature, error) {
+	upwardsFeatures := make(map[string]*featureproto.Feature, 0)
+	for _, target := range targetFeatures {
+		for _, newTarget := range featuresHavePrerequisite {
+			for _, p := range newTarget.Prerequisites {
+				if p.FeatureId == target.Id {
+					if _, ok := upwardsFeatures[newTarget.Id]; ok {
+						continue
+					}
+					upwardsFeatures[newTarget.Id] = newTarget
+				}
+			}
+		}
+	}
+	if len(upwardsFeatures) == 0 {
+		return targetFeatures, nil
+	}
+	newTargets, err := s.getPrerequisiteUpwards(upwardsFeatures, featuresHavePrerequisite)
+	if err != nil {
+		return nil, err
+	}
+	return s.mapMerge(targetFeatures, newTargets), nil
+}
```

Comment:
Same as https://github.com/bucketeer-io/bucketeer/pull/284/files#r1166773359.

```suggestion
/*
getPrerequisiteUpwards gets the features that have the specified targetFeatures as the prerequisite.
*/
func (s *grpcGatewayService) getPrerequisiteUpwards( // nolint:unused
	targetFeatures, featuresHavePrerequisite map[string]*featureproto.Feature,
) (map[string]*featureproto.Feature, error) {
	upwardsFeatures := make(map[string]*featureproto.Feature, 0)
	// depth first search
	queue := make([]*featureproto.Feature, 0)
	for _, f := range targetFeatures {
		queue = append(queue, f)
	}
	for len(queue) > 0 {
		f := queue[0]
		for _, newTarget := range featuresHavePrerequisite {
			for _, p := range newTarget.Prerequisites {
				if p.FeatureId == f.Id {
					if _, ok := upwardsFeatures[newTarget.Id]; ok {
						continue
					}
					upwardsFeatures[newTarget.Id] = newTarget
					queue = append(queue, newTarget)
				}
			}
		}
		queue = queue[1:]
	}
	if len(upwardsFeatures) == 0 {
		return targetFeatures, nil
	}
	return s.mapMerge(targetFeatures, upwardsFeatures), nil
}
```